cmd/update-main.go

```diff
@@ -226,11 +240,36 @@ func getLatestReleaseTime(timeout time.Duration, mode string) (releaseTime time.
 	return parseReleaseData(data)
 }
 
-func getDownloadURL() (downloadURL string) {
+// Converts build version date into a release tag
+// format, relPrefix (release prefix) string is
+// optional but mostly provided in all cases.
+func releaseTag(version, relPrefix string) string {
+	if version == "" {
+		return ""
+	}
+	relTag := strings.Replace(version, " ", "-", -1)
+	relTag = strings.Replace(relTag, ":", "-", -1)
+	relTag = strings.Replace(relTag, ",", "", -1)
+	if relPrefix != "" {
+		return relPrefix + "." + relTag
+	}
+	return relTag
+}
+
+func getDownloadURL(buildDate time.Time) (downloadURL string) {
+	rTag := releaseTag(buildDate.Format(time.RFC3339), "RELEASE")
```

Comment:
you could use `minioReleaseTagTimeLayout`.  IMO `releaseTag()` is not required as separate function.